lib/recommendation-server.js

```diff
@@ -31,6 +31,8 @@ function RecommendationServer(opts) {
                 this.prefs.getCharPref('server') :
                 'http://universal-search.dev';
   this.search = this.search.bind(this);
+  this.prev = null;
+  this._resultHasChanged.bind(this);
```

Comment:
Here, we technically want to assign the result of binding to the object, so that we can both attach and detach the same bound listener:

`this._resultHasChanged = this._resultHasChanged.bind(this)`

Naming is hard. I might go with `isNewRecommendation` or `matchesCurrent`, putting the 'has' in the middle seems a bit odd.